Mathlib/Data/Int/Cast/Basic.lean

```diff
@@ -95,13 +95,13 @@ theorem cast_negOfNat (n : ℕ) : ((negOfNat n : ℤ) : R) = -n := by simp [Int.
 @[simp, norm_cast]
 theorem cast_add : ∀ m n, ((m + n : ℤ) : R) = m + n
   | (m : ℕ), (n : ℕ) => by simp [← Int.natCast_add]
-  | (m : ℕ), -[n+1] => by erw [cast_subNatNat, cast_natCast, cast_negSucc, sub_eq_add_neg]
-  | -[m+1], (n : ℕ) => by
-    #adaptation_note
-    /-- `_root_` can be removed again after
-    https://github.com/leanprover/lean4/pull/7359 lands in nightly-2025-03-06. -/
-    erw [cast_subNatNat, cast_natCast, cast_negSucc, _root_.sub_eq_iff_eq_add, add_assoc,
-      eq_neg_add_iff_add_eq, ← Nat.cast_add, ← Nat.cast_add, Nat.add_comm]
+  | (m : ℕ), -[n+1] =>
+    show (subNatNat m n.succ : R) = _ by
```

Comment:
I'm not convinced `show` is better than `erw` here: both are hiding what might be hidden problems in mathlib.
Can you either say why you think this is fine, or remove the erw changes? Thanks!

Reply:
This is fine because the fourth case below also uses `show`.